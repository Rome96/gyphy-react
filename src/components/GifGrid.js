import React from 'react'
import PropTypes from 'prop-types'
import GifGridItem from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

const GifGrid = ({category}) => {
  const {loading, data:images} = useFetchGifs(category)
  return <>
    <p className="title__category animate__animated animate__backInDown animate__delay-1s">{category}</p>
      { loading && <p className="animate__animated animate__flash">Cargando...</p> }
      <div className='card-grid'>
        {images.map(img => (
          <GifGridItem
            key={img.id}
            {...img}
          />
        ))}
      </div>
  </>
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export default GifGrid
