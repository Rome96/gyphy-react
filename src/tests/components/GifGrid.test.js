import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import GifGrid from '../../components/GifGrid';

describe('Testing in component <GifGrid/>', () => {
  const category = 'Goku';
  let wrapper = shallow(<GifGrid category={category} />);

  beforeEach(() => {
    wrapper = shallow(<GifGrid category={category} />);
  });

  test('Render component  <GifGrid/>', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
});
