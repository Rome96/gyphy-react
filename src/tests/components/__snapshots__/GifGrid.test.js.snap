// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component <GifGrid/> Render component  <GifGrid/> 1`] = `
<Fragment>
  <p
    className="title__category animate__animated animate__backInDown animate__delay-1s"
  >
    Goku
  </p>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
